test/partial-fulfill.spec.ts

```diff
@@ -320,6 +507,142 @@ describeWithFixture(
+              (item) => ({
+                ...item,
+                token,
+              }),
+            ),
+          };
+
+          await testErc20.mint(
+            await fulfiller.getAddress(),
+            (standardCreateOrderInput.consideration[0] as CurrencyItem).amount,
+          );
+
+          const { executeAllActions } = await seaport.createOrder(
+            standardCreateOrderInput,
+          );
+
+          const order = await executeAllActions();
+
+          expect(order.parameters.orderType).eq(OrderType.PARTIAL_OPEN);
+
+          const orderStatus = await seaport.getOrderStatus(
+            seaport.getOrderHash(order.parameters),
+          );
+
+          const ownerToTokenToIdentifierBalances =
+            await getBalancesForFulfillOrder(
+              order,
+              await fulfiller.getAddress(),
+            );
+
+          const { actions } = await seaport.fulfillOrder({
+            order,
+            unitsToFill: 2,
+            accountAddress: await fulfiller.getAddress(),
+            domain: OPENSEA_DOMAIN,
+            tips,
+          });
+
+          expect(actions.length).to.eq(2);
+
+          const approvalAction = actions[0];
+
+          expect(approvalAction).to.deep.equal({
+            type: "approval",
+            token: await testErc20.getAddress(),
+            identifierOrCriteria: "0",
+            itemType: ItemType.ERC20,
+            transactionMethods: approvalAction.transactionMethods,
+            operator: await seaport.contract.getAddress(),
+          });
+
+          await approvalAction.transactionMethods.transact();
+
+          expect(
+            await testErc20.allowance(
+              await fulfiller.getAddress(),
+              await seaport.contract.getAddress(),
+            ),
+          ).to.eq(MAX_INT);
+
+          const fulfillAction = actions[1];
+
+          expect(fulfillAction).to.be.deep.equal({
+            type: "exchange",
+            transactionMethods: fulfillAction.transactionMethods,
+          });
+
+          const { data } =
+            await fulfillAction.transactionMethods.buildTransaction();
+
+          const iface = new Interface(SeaportABIv15);
+          const decoded = iface.parseTransaction({ data });
+          const considerations = decoded?.args[0][0][3];
+          expect(considerations.length).to.eq(3);
+
+          const tipConsideration = considerations.find(
+            (consideration: {
+              recipient: string;
+              token: string;
+              startAmount: BigInt;
+            }) =>
+              consideration.recipient === tips[0].recipient &&
+              consideration.token === tips[0].token &&
+              consideration.startAmount === BigInt(tips[0].amount),
+          );
+          // This test verifies that the tip consideration exists.
+          // Tip considerations will be adjusted correctly by the seaport protocol.
```

Comment:
note: ERC20 payments were never impacted by the bug I've discovered. 

In the case of an ERC20 payment since the native ETH value is always 0. Here, I'm checking that the tip consideration is included correctly. And of course, I check the balances after fulfilling the order.